GitCommands/ExceptionUtils.cs

```diff
@@ -8,6 +8,41 @@ namespace GitCommands
 {
     public static class ExceptionUtils
     {
+        /// <summary>
+        /// Determines whether the <paramref name="ex"/> contains an inner exception of the desired type <typeparamref name="T"/>.
+        /// </summary>
+        /// <typeparam name="T">Desired exception type.</typeparam>
+        /// <param name="ex">The exception to inspect.</param>
+        /// <returns><see langword="true"/> if one of the inner exceptions are of desired type; otherwise <see langword="false"/>.</returns>
+        public static bool HasInnerOfType<T>(this Exception ex)
+           where T : Exception
+        {
+            return ex.InnerOfType<T>() != null;
+        }
+
+        /// <summary>
+        /// Determines whether the <paramref name="ex"/> contains an inner exception of the desired type <typeparamref name="T"/> and returns it, if found.
+        /// </summary>
+        /// <typeparam name="T">Desired exception type.</typeparam>
+        /// <param name="ex">The exception to inspect.</param>
+        /// <returns>The inner exceptions are of desired type; otherwise <see langword="null"/>.</returns>
+        public static T InnerOfType<T>(this Exception ex)
```

Comment:
Should this also handle `AggregateException.InnerExceptions`?

Reply:
Good idea. 
I will do it later as a separate submission as it is not a blocker.